python/pyarrow/__init__.py

```diff
@@ -252,7 +252,7 @@ def print_entry(label, value):
                          create_memory_map, MockOutputStream,
                          input_stream, output_stream)
 
-from pyarrow._hdfsio import HdfsFile, have_libhdfs
+from pyarrow._hdfsio import have_libhdfs
```

Comment:
It seems we forgot to explicitly deprecate accessing HdfsFile, but I guess everybody doing that would also access the filesystem itself, and should have seen the warning.

I suppose `have_libhdfs` still has its use for the non-legacy pyarrow.fs.HadoopFileSystem (eg we use it in their tests)